### PR TITLE
[single] More inline functions for flexible dimensions

### PR DESCRIPTION
**Pull request by kparichay**
Added extra inline funcitons to provide easier API for dyanmic input dimensions

Check #1651 for more information

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: api/capi/include/nnstreamer-single.h [hunk 1]
```diff
@@ -106,6 +106,26 @@ int ml_single_close (ml_single_h single);
  */
 int ml_single_invoke (ml_single_h single, const ml_tensors_data_h input, ml_tensors_data_h *output);
 
+/**
+ * @brief Invokes the model with the given input data with the given info.
+ * @details This function changes the input data dimensions for the model.
+ *          Note that this has a default timeout of 3 seconds. If an application wants to change the time to wait for an output, set the timeout using ml_single_set_timeout().
+ * @param[in] single The model handle to be inferred.
```
Comment (kparichay) [inline api/capi/include/nnstreamer-single.h]: @myungjoo @again4you how should `@since_tizen` be described here? I have currently omitted it.

Comment (myungjoo) [reply api/capi/include/nnstreamer-single.h]: It'd be ```@since_tizen 6.0```

File: api/capi/include/nnstreamer-single.h [hunk 2]
```diff
@@ -163,6 +197,52 @@ int ml_single_set_input_info (ml_single_h single, const ml_tensors_info_h info);
...
+
+  status = ml_single_update_info (single, in_info, out_info);
+  if (status != 0)
+    return status;
+
+  status = ml_single_invoke (single, input, output);
+  if (status != 0) {
+    ml_single_set_input_info (single, cur_in_info);
+    ml_tensors_data_destroy (out_info);
+  }
```
Comment (myungjoo) [inline api/capi/include/nnstreamer-single.h]: replace return value 0 with the macro

File: api/capi/include/nnstreamer-single.h (hunk 1 above)
Comment (kparichay) [reply api/capi/include/nnstreamer-single.h]: updated

File: api/capi/include/nnstreamer-single.h (hunk 2 above)
Comment (kparichay) [reply api/capi/include/nnstreamer-single.h]: that my mistake, updated.

File: api/capi/include/nnstreamer-single.h [hunk 3]
```diff
@@ -163,6 +200,52 @@ int ml_single_set_input_info (ml_single_h single, const ml_tensors_info_h info);
...
+    return status;
+
+  status = ml_single_update_info (single, in_info, out_info);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  status = ml_single_invoke (single, input, output);
+  if (status != ML_ERROR_NONE) {
+    ml_single_set_input_info (single, cur_in_info);
+    ml_tensors_data_destroy (out_info);
```
Comment (jaeyun-jung) [inline api/capi/include/nnstreamer-single.h]: maybe ml_tensors_info_destroy (*out_info);

File: api/capi/include/nnstreamer-single.h [hunk 4]
```diff
@@ -106,6 +106,27 @@ int ml_single_close (ml_single_h single);
...
+ * @param[out] out_info The handle of output tensors information. The caller is responsible for freeing the information with ml_tensors_info_destroy().
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_STREAMS_PIPE Cannot push a buffer into source element.
+ * @retval #ML_ERROR_TIMED_OUT Failed to get the result from sink element.
+ *
+ */
+static inline int ml_single_invoke_dynamic (ml_single_h single, const ml_tensors_data_h input, const ml_tensors_info_h in_info,
```
Comment (jaeyun-jung) [inline api/capi/include/nnstreamer-single.h]: [Question] why the inline function is declared for dynamic tensor shape?

Comment (myungjoo) [reply api/capi/include/nnstreamer-single.h]: We want to provide some syntactic sugar for developer's easier development.

Risk: The API team may hate it and tell us to hide the function budy, making it regular function.

**Comment by myungjoo** [comment]
Note that we are going to process "ACR" with this (and other changes if available) after Tizen 5.5 release is completed.

File: api/capi/include/nnstreamer-single.h (hunk 3 above)
Comment (kparichay) [reply api/capi/include/nnstreamer-single.h]: aah, updated.

**Comment by myungjoo** [comment]
PTAL other Tizen API experts: @helloahn @again4you

**Pull request merged**